dpx/src/dpx_pdfobj.rs

```diff
@@ -1281,14 +1280,10 @@ impl pdf_dict {
     /* pdf_merge_dict makes a link for each item in dict2 before stealing it */
     pub unsafe fn merge(&mut self, dict2: &Self) {
         for (k, &v) in dict2.inner.iter() {
-            let linked = pdf_link_obj(v);
-            self.inner
-                .entry(k.clone())
-                .and_modify(|existing| {
-                    let existing = mem::replace(existing, linked);
-                    pdf_release_obj(existing);
```

Comment:
Releasing is important, right?

Reply:
I think no, not here.

Reply:
1. [The old code does it](https://github.com/crlf0710/tectonic/pull/212/files#diff-783cb601464af70fc34a0609b6535172L1272)
2. Indexmap does no Drop with raw pointers, so it is still necessary to match the old code
3. The codebase treats dicts as owning the values and doesn't free them after adding them to the dict
4. I don't see anything special about merging that would not require releasing the previous object when deleting and draining do require it

Reply:
On the other hand, using `Rc` will steamroll this eventually.

Reply:
`merge` use `pdf_dict`s `set` which release old value